### PR TITLE
Fix TaskGroup cannot work and will cause master dead loop

### LINKED ISSUE #11253: [Bug] [Master] TaskGroup cannot work and will cause master handle event dead loop.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

I find when we create a workflow with TaskGroup, the workflow may go into dead loop, and status will still in running.

### What you expected to happen

work well

### How to reproduce

1. Create a TaskGroup with 3 capaticy.
2. Create a workflow with 6 task, and each task has a unique priority, e.g 1, 2, 3, 4, 5, 6, and use the task group
3. Execute the workflow, find the workflow instance is always running and the master handle event task state change event dead loop, cause by release task group deap loop.

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by ruanwenjun**
## Purpose of the pull request

close #11253 

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java [hunk 1]
```diff
@@ -2965,7 +2967,7 @@ public boolean robTaskGroupResouce(TaskGroupQueue taskGroupQueue) {
 
     @Override
     public boolean acquireTaskGroupAgain(TaskGroupQueue taskGroupQueue) {
-        return robTaskGroupResouce(taskGroupQueue);
+        return robTaskGroupResource(taskGroupQueue);
```
Comment (caishunfeng) [inline dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java]: I think the method `acquireTaskGroupAgain` can be removed and use `robTaskGroupResouce` directly.

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java [hunk 2]
```diff
@@ -2984,40 +2986,41 @@ public void releaseAllTaskGroup(int processInstanceId) {
...
-        if (thisTaskGroupQueue.getStatus() == TaskGroupQueueStatus.RELEASE) {
-            return null;
-        }
+        TaskGroup taskGroup;
+        TaskGroupQueue thisTaskGroupQueue;
         try {
-            while (thisTaskGroupQueue.getForceStart() == Flag.NO.getCode() && taskGroupMapper.releaseTaskGroupResource(taskGroup.getId(), taskGroup.getUseSize()
-                , thisTaskGroupQueue.getId(), TaskGroupQueueStatus.ACQUIRE_SUCCESS.getCode()) != 1) {
+            do {
+                taskGroup = taskGroupMapper.selectById(taskInstance.getTaskGroupId());
```
Comment (caishunfeng) [inline dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java:2988]: Is it necessary to query task group looply?

Comment (ruanwenjun) [reply dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java:2988]: The use size will change.

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java (hunk 1 above)
Comment (ruanwenjun) [reply dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java]: Done

**Review by caishunfeng — APPROVED** [review]
LGTM

**Pull request merged**